Add escape sequences and test

ihmels opened the pull request:
I think, the backslashes are not needed.

ljharb commented:
They are in fact needed; see https://github.com/creationix/nvm/pull/1322#discussion_r89441322, #1279, and #1278.

ljharb commented:
We should add a regression test that aliases `.`, so that these tests would have failed. Want to update this PR to do that instead?

ihmels commented:
I updated my pull request to only remove the backslashes in README. I also added a test, in which an alias for `.` is set, but unfortunately it was not executed on Travis.

ljharb commented on test/sourcing/Sourcing nvm.sh with alias for . should succeed:
```diff
@@ -0,0 +1,9 @@
+
+alias .='cat'
+
+\. ../../nvm.sh
```
the `\.` here will make the alias not do anything; we don't need to test `.` vs `\.`, what we need to test is the *install script* (not sourcing).

ihmels replied:
So I have to define the alias in all files under `test/install_script/`, so that `install.sh` would always fail, if `.` is not escaped?

ljharb replied:
not necessarily in all files, a single test would be better.

ljharb commented on README.md:
```diff
@@ -135,8 +135,8 @@ Now add these lines to your `~/.bashrc`, `~/.profile`, or `~/.zshrc` file to hav
 export NVM_DIR="$HOME/.nvm"
-[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
-[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"  # This loads nvm
```
People will copy-paste this code; it's important it retains the `\`.

ihmels replied:
But there are lines in the README, where no `\` has been placed (see section “Install script”). These lines have to be modified, too.

ljharb replied:
ah, that's an oversight then. anything in a code block should have the `\`.

ihmels commented:
I’ve updated my pull request. Do I need to add the test file to something? The new test is not executed on Travis.

ljharb commented:
Yes, you'll need to `git add` it, and also `chmod a+x` it :-) (also, maybe use "dot" instead of "." in the filename, just in case)

ljharb commented on test/install_script/nvm_install_with_aliased_dot:
```diff
@@ -1,13 +1,15 @@
+  shopt -s expand_aliases
   alias .='cat'
-  NVM_ENV=testing \. ../../install.sh
+  \. ../../install.sh > /dev/null
```
the `NVM_ENV=testing` is still required tho; otherwise it will actually invoke the install script.

ihmels replied:
But when I set `NVM_ENV=testing`, `nvm_do_install` is not executed, so I can’t test the sourcing.

ljharb replied:
You can test the sourcing without executing that function; for example, `type nvm_do_install` will only exit zero if it was sourced correctly.

ihmels replied:
I want to test line 368 in `install.sh`. `type nvm_do_install` is not enough. Or do you have another idea?

ljharb replied:
ah, true. but in that case, you can invoke `nvm_do_install` manually.

ihmels replied:
I think I'm done.
